TVHeadEnd/AccessTicketHandler.cs

```diff
@@ -0,0 +1,116 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using TVHeadEnd.HTSP;
+using TVHeadEnd.HTSP_Responses;
+using TVHeadEnd.TimeoutHelper;
+
+namespace TVHeadEnd;
+
+public class AccessTicketHandler
+{
+    private readonly ILogger<AccessTicketHandler> _logger;
+
+    private readonly HTSConnectionHandler _htsConnectionHandler;
+    private readonly string _ticketItemType;
+    private readonly TimeSpan _requestTimeout;
+    private readonly TimeSpan _ticketLifeSpan;
+
+    private volatile int _ticketIdSequence;
+
+    public enum TicketType : byte { Channel, Recording };
+
+    public record Ticket
+    {
+        public string Id { get; init; }
+        public string Path { get; init; }
+        public string TicketParam { get; init; }
+        public DateTime Expires { get; init; }
+    }
+
+    private readonly ConcurrentDictionary<string, Ticket> _ticketCache = new();
+
+    internal AccessTicketHandler(
+        ILoggerFactory loggerFactory, HTSConnectionHandler htsConnectionHandler,
+        TimeSpan requestTimeout, TimeSpan ticketLifeSpan, TicketType ticketType)
+    {
+        _logger = loggerFactory.CreateLogger<AccessTicketHandler>();
+        _htsConnectionHandler = htsConnectionHandler;
+        _requestTimeout = requestTimeout;
+        _ticketLifeSpan = ticketLifeSpan;
+
+        _ticketItemType = ticketType switch
+        {
+            TicketType.Channel => "channelId",
+            TicketType.Recording => "dvrId",
+            _ => throw new ArgumentException("undefined ticketType")
+        };
+    }
+
+    public async Task<Ticket> GetTicket(string itemId, CancellationToken cancellationToken)
+    {
+        var now = DateTime.Now;
```

Comment:
```suggestion
        var now = DateTime.UtcNow;
```
We typically prefer UTC unless required to be local

Reply:
No problem. I'll update it.